internal/services/applications/application_resource.go

```diff
@@ -57,21 +53,101 @@ func applicationResource() *schema.Resource {
+				Deprecated:       "This property has been renamed to `display_name` and will be removed in version 2.0 of the AzureAD provider",
 				ExactlyOneOf:     []string{"display_name", "name"},
 				ValidateDiagFunc: validate.NoEmptyStrings,
 			},
 
+			"api": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// TODO: v2.0 also consider another computed typemap attribute `oauth2_permission_scope_ids` for easier consumption
+						"oauth2_permission_scope": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Computed: true,
+							Set:      tf.SetFuncHashId,
+							// SchemaConfigModeAttr seems to break d.Get("api.0.oauth2_permission_scope") ??
+							// Maybe re-enable in v2.0 when we'll be using the expand func for `api`
+							// (The goal being to enable config to set `oauth2_permission_scope = []`)
+							//ConfigMode: schema.SchemaConfigModeAttr,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+
+									"admin_consent_description": {
+										Type:             schema.TypeString,
+										Optional:         true,
+										ValidateDiagFunc: validate.NoEmptyStrings,
+									},
+
+									"admin_consent_display_name": {
+										Type:             schema.TypeString,
+										Optional:         true,
+										ValidateDiagFunc: validate.NoEmptyStrings,
+									},
+
+									"enabled": {
+										Type:     schema.TypeBool,
+										Optional: true,
+									},
+
+									// TODO: v2.0 remove this
+									"is_enabled": {
+										Type:     schema.TypeBool,
+										Computed: true,
+									},
+
+									"type": {
+										Type:     schema.TypeString,
+										Optional: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(msgraph.PermissionScopeTypeAdmin),
+											string(msgraph.PermissionScopeTypeUser),
+										}, false),
+									},
+
+									"user_consent_description": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+
+									"user_consent_display_name": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+
+									"value": {
+										Type:             schema.TypeString,
+										Optional:         true,
+										ValidateDiagFunc: validate.NoEmptyStrings,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			// TODO: v2.0 consider another computed typemap attribute `app_role_ids` for easier consumption
 			"app_role": {
 				Type:       schema.TypeSet,
 				Optional:   true,
 				Computed:   true,
 				ConfigMode: schema.SchemaConfigModeAttr,
+				//Set:        tf.SetFuncHashId, // TODO: v2.0 enable this SchemaSetFunc when id is no longer Computed
```

Comment:
we probably don't want to do this since it means changes to fields other than ID won't be tracked?

Reply:
Absolutely. I misunderstood the semantics of changes to set members, this doesn't do what I had expected 👍